scripts/ci.sh

```diff
@@ -22,7 +22,8 @@ run_geth() {
 		--dev \
 		--dev.period 0 \
 		--allow-insecure-unlock \
-		js ./scripts/geth-accounts.js \
+		--preload ./scripts/geth-accounts.js \
+		console
```

Comment:
The hypothesis is without the pseudo-terminal provided by `-t`, GETH's console will not receive a stdin and triggered the full process exit.

Reply:
> The hypothesis is without the pseudo-terminal provided by `-t`, GETH's console will not receive a stdin and somehow trigger the full process exit.

FWIW, I'm quite confident that this what's happening. The interactive terminal session ran by geth behaves much like most other interactive terminals and terminates immediately upon `EOF`. Unless you explicitly tell docker to allocate a psuedo-tty for your process by passing the `-t` flag, the process behaves as though `stdin` was redirected. That means that once the process has read the contents of whatever was redirected to `stdin` (e.g. like a file redirect, pipe, etc), further calls to `read` against the `stdin` file handle will always return `0` rather than blocking on new data, triggering the `EOF` condition.